src/links.ts

```diff
@@ -0,0 +1,61 @@
+import { PluginExtensionAddedLinkConfig, PluginExtensionPoints, RawTimeRange } from '@grafana/data';
+import { DataQuery } from '@grafana/schema';
+import { GrafanaPyroscopeDataQuery } from '@grafana/schema/dist/esm/raw/composable/grafanapyroscope/dataquery/x/GrafanaPyroscopeDataQuery_types.gen';
+
+export type PluginExtensionExploreContext = {
+  targets: DataQuery[];
+  timeRange: RawTimeRange;
+};
+
+type URLParamsBuilderProps = {
+  pyroscopeQuery: GrafanaPyroscopeDataQuery;
+  timeRange?: RawTimeRange;
+};
+
+function buildURL(props: URLParamsBuilderProps) {
+  const { timeRange, pyroscopeQuery } = props;
+
+  let timeRangeParam = '';
+  let explorationType = 'all';
+
+  let serviceName = props.pyroscopeQuery.labelSelector?.match(/service_name="([^"]+)"/)?.[1];
+
+  if (serviceName) {
+    explorationType = 'labels';
+  }
+
+  const datasourceParam = `var-dataSource=${pyroscopeQuery.datasource?.uid}`;
+  const serviceNameParam = serviceName ? `&var-serviceName=${serviceName}` : '';
+  const profileTypeParam = `&var-profileMetricId=${pyroscopeQuery.profileTypeId}`;
+  const explorationTypeParam = `&explorationType=${explorationType}`;
+  if (timeRange) {
+    timeRangeParam = `&from=${timeRange.from}&to=${timeRange.to}`;
+  }
+
+  return `/a/grafana-pyroscope-app/profiles-explorer?${datasourceParam}${serviceNameParam}${profileTypeParam}${timeRangeParam}${explorationTypeParam}`;
```

Comment:
Maybe it's safer to use https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams ?